### PR TITLE
Add support for reusing common configuration between all application …

### PR DESCRIPTION
**Pull request by igordumchykov**
…in scope of one tenant

### CODE REVIEW

**Review by VictorKhoroshko — APPROVED** [review]

File: xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/XmLepScriptConfigServerResourceLoader.java [hunk 1]
```diff
@@ -31,6 +31,12 @@
     // /config/tenant/{tenant-key}/{ms-name}/lep/**
     private final String tenantLepScriptsAntPathPattern;
 
+    // /config/tenant/{tenant-key}/commons/lep/**
+    private static final String commonsLepScriptsAntPathPattern = "/config/tenants/{tenantKey}/commons/lep/**";
+
+    // /config/tenant/commons/lep/**
+    private static final String environmentLepScriptsAntPathPattern = "/config/tenants/commons/lep/**";
```
Comment (amedvedchuk) [inline xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/XmLepScriptConfigServerResourceLoader.java:38]: Seems from now it is not allowed now to create tenant with the name `commons`.

Suggest checking the following places where pattern is used:
1. `com.icthh.xm.commons.tenant.TenantContextUtils.TENANT_KEY_REGEXP`
2. `resources/swagger-spec/tenant/swagger-tenant-spec.yaml`  line 145
3. `com.icthh.xm.commons.tenant.TenantKeyUnitTest#testValidateTenantKey`

Comment (igordumchykov) [reply xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/XmLepScriptConfigServerResourceLoader.java:38]: Done:
- refactor TenantScriptStorage according to https://github.com/xm-online/xm-commons/tree/feature-add-support-commons-for-all-apps-refactor
- exclude commons tenant name from TENANT_KEY_REGEXP
- add unit tests on tenant name exclusion

File: xm-commons-lep/src/test/java/com/icthh/xm/commons/lep/TenantScriptStorageUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Test
+    public void testResolveLepForClassPathScriptStorage() {
+        String envLepPath = CLASSPATH.resolvePath(TENANT_KEY, APP_NAME, URL_PREFIX_COMMONS_ENVIRONMENT + PATH);
+        String tenantLepPath = CLASSPATH.resolvePath(TENANT_KEY, APP_NAME, URL_PREFIX_COMMONS_TENANT + PATH);
+        String localLepPath = CLASSPATH.resolvePath(TENANT_KEY, APP_NAME, PATH);
+
+        assertEquals("classpath:/lep/custom/commons/functions/aggregation", envLepPath);
+        assertEquals("classpath:/lep/custom/xm/commons/functions/aggregation", tenantLepPath);
+        assertEquals("classpath:/lep/custom/xm/commons/functions/aggregation", localLepPath);
```
Comment (amedvedchuk) [inline xm-commons-lep/src/test/java/com/icthh/xm/commons/lep/TenantScriptStorageUnitTest.java:27]: @igordumchykov is it correct that these two lines above are the same? Seems localLepPath should contain the application name.

**Review by amedvedchuk — APPROVED** [review]

File: xm-commons-lep/src/test/java/com/icthh/xm/commons/lep/TenantScriptStorageUnitTest.java (hunk 2 above)
Comment (RomanTeren) [reply xm-commons-lep/src/test/java/com/icthh/xm/commons/lep/TenantScriptStorageUnitTest.java]: I was wrong. These changes failed tests, fix: https://github.com/xm-online/xm-commons/pull/119

**Pull request merged**